src/Rings/mpoly-localizations.jl

```diff
@@ -1673,13 +1702,23 @@ function saturated_ideal(
           end
         end
         if length(cache) > 0
-          extend_pre_saturated_ideal!(I, 
-                                      elem_type(R)[g for (g, x, u) in cache],
-                                      vcat(dense_matrix_type(R)[x for (g, x, u) in cache]),
-                                      [u for (g, x, u) in cache], 
-                                      check=false
-                                     )
+          # We completely overwrite the pre_saturated_ideal with the generators 
+          # of the saturated ideal
+          A = zero(MatrixSpace(L, ngens(Jsat), ngens(I)))
+          for i in 1:length(cache)
+            (g, a, dttk) = cache[i]
+            A[i, :] = L(one(dttk), dttk, check=false)*change_base_ring(L, a)*pre_saturation_data(I)
+          end
+          I.pre_saturated_ideal = Jsat
+          I.pre_saturation_data = A
```

Comment:
This needs a test.

Reply:
Then I would merge it :-).

Reply:
> Then I would merge it

But with a squash, please. It's become too messy.